src/scripts/main.js

```diff
@@ -1,3 +1,221 @@
+  </label>
+  <label>
+    Salary: <input name="salary" type="number" data-qa="salary" required>
+  </label>
+  <button type="button">Save to table</button>
+`;
+
+body.append(form);
+
+form.querySelector('button').addEventListener('click', (e) => {
+  // e.preventDefault();
+
+  if (!checkFormFields()) {
+    return;
+  }
+  
+  const newRow = tBody.insertRow();
+
+  if (
+    form.elements.name.value &&
+    form.elements.position.value &&
+    form.elements.office.value &&
+    form.elements.age.value &&
+    form.elements.salary.value
+  ) {
+      newRow.insertCell(0).textContent = form.elements.name.value.trim();  
+      newRow.insertCell(1).textContent = form.elements.position.value.trim();
+      newRow.insertCell(2).textContent = form.elements.office.value.trim();
+      newRow.insertCell(3).textContent = parseInt(form.elements.age.value, 10);
+      let salaryFormated = parseFloat(form.elements.salary.value.replace(/[^0-9.]/g, ''));
+      newRow.insertCell(4).textContent = `$${new Intl.NumberFormat("us-IN").format(salaryFormated)}`;
+  }
+
+  checkFormFields();
+  tBody.append(newRow);
+  
+  showNotification(
+    'Success!',
+    'New employee is successfuly added',
+    'success'
+  )
+
+  form.reset();
+});
+
+function showNotification(title, message, type) {
+  const notification = document.createElement('div');
+
+  notification.className = `notification ${type}`;
+  notification.setAttribute('data-ga', 'notification');
+
+  notification.innerHTML = `
+    <h2 class="title">${title}</h2>
+    <p>${message}</p>
+  `;
+
+  document.body.append(notification);
+
+  setTimeout(() => {
+    notification.remove();
+  }, 4000);
+}
+
+function checkFormFields() {
+  const employeeName = form.elements.name.value.trim();
+
+  if (employeeName.length > 0 && employeeName.length < 4) {
+    showNotification(
+      'Error!',
+      'The minimum name length is 4 letters',
+      'error'
+    );
+
+    return false;
+  }
+
+  const employeeAge = parseInt(form.elements.age.value, 10);
+
+  if (employeeAge < 18 || employeeAge > 90) {
+    showNotification(
+      'Error!',
+      'You are too young/old for this',
+      'error'
+    )
+
+    return false;
+  }
+
+  [...form.elements].every(input => {
+    if (input.tagName === 'INPUT' || input.tagName === 'SELECT') {
+      if (!input.value.trim()) {
+        showNotification(
+          'Error!',
+          'Fill in all fields',
+          'error'
+        )
+
+        return false;
+      }
+    }
+
+    return true;
+  });
+
+  return true;
+}
+
+//Making editable table content:
+
+cells.forEach(cell => {
+  const editCell = document.createElement('input');
+  editCell.classList.add('cell-input');
+  
+  cell.addEventListener('dblclick', () => {
+    cell.replaceWith(editCell);
+  });
+
+  editCell.addEventListener('keypress', (e) => {
+    if (e.key === 'Enter') {
+      editCell.replaceWith(td);
+      td.textContent = editCell.value;
```

Comment:
The logic for replacing the editable cell with `td` is incorrect. `td` is a single element created outside the loop, which means it will be reused for all cells. Instead, create a new `td` element inside the event listener to ensure each cell has its own element.